docs: replace grafana-agent with opentelemetry-collector (#258)
canonical/observability-stack: docs/explanation/architecture/telemetry-flow.md

COS Lite includes Loki, a logs backend, and Prometheus, a metrics backend. The API endpoints of Loki and Prometheus are communicated to other charms via Juju integrations. This way, telemetry producers (or aggregators) are able to push telemetry to Loki and Prometheus.

> Prometheus is also able to pull metrics directly from metric providers. However, this is not generally advisable over model or substrate boundaries. In the recommended deployment scenarios, a telemetry aggregator, such as Opentelemetry Collector, is used to funnel logs and metrics to COS.

Reviewer: ```suggestion
Prometheus is also able to pull metrics directly from metric providers. However, this is not generally advisable over model or substrate boundaries. In the recommended deployment scenarios, a telemetry aggregator, such as Opentelemetry Collector, is used to funnel telemetry to COS.
```

> In this case, Opentelemetry Collector takes over the responsibility to push logs as well as metrics into COS: so it’s not required for the COS stack to be able to reach the Opentelemetry Collector, but the agent must be able to reach the COS stack.

Reviewer: ```suggestion
In this case, Opentelemetry Collector takes over the responsibility to push telemetry into COS: so it’s not required for the COS stack to be able to reach the Opentelemetry Collector, but the agent must be able to reach the COS stack.
```

The COS Proxy bridges the gap between workloads instrumented using NRPE, allowing it to be turned into metrics stored as time series in Prometheus. This charm is meant to serve as a stepping stone, and not as a future-proof solution. Going forward, all charms built by Canonical will instead expose metrics directly, allowing users to omit the proxy.

### Opentelemetry Collector

The Opentelemetry Collector machine charm aims to collect both host and application telemetry in deployment scenarios where a full virtual machine is being used. It is powered by the `opentelemetry-collector` snap, and handed a configuration containing all scrape targets available on the machine, including a `node_exporter` endpoint provided by the snap itself.

### Opentelemetry Collector K8s

> The Opentelemetry Collector K8s charm is analogous to the machine ditto but deployed in a K8s pod. In contrast to the machine charm, the K8s charm is not installed in the same container as the workload, which means scraping or log forwarding happens between pods rather than in-pod.

Reviewer: ```suggestion
The Opentelemetry Collector K8s charm is analogous to the machine charm, but deployed in a K8s pod. In contrast to the machine charm, the K8s charm is not installed in the same container as the workload, which means scraping or log forwarding happens between pods rather than in-pod.
```

### Prometheus Scrape Config

`prometheus-scrape-config` is a workload-less charm used to alter the configuration of individual scrape jobs. It is commonly used to modify the scrape interval of the cos-proxy NRPE checks, as triggering an involved bash script once every minute has the potential of degrading system performance.

> In a typical OpenStack + COS deployment, there are multiple of these with different scrape intervals, depending on the part of the system you are monitoring.

Reviewer: ```suggestion
In a typical OpenStack + COS deployment, there are multiple of these charms with different scrape intervals, depending on the part of the system you are monitoring.
```
